### PR TITLE
DAOS-8655 tests: Fix orterun tmpdir 'No such file or directory'

### PR DESCRIPTION
**Pull request by phender**
Use a test-specific temporary directory for temporary files generated
during the test and remove the files after test completion to free up
space for the next test.

Quick-functional: true
Test-tag: metadata_ior offline_extend_oclass
Test-repeat: 10

Signed-off-by: Phillip Henderson phillip.henderson@intel.com

### CODE REVIEW

**Review by sylviachanoiyee — COMMENTED** [review]
Quite a bit of changes in the common files.
Let's kick off a run with default testlist to make sure nothing breaks.

File: src/tests/ftest/server/metadata.py [hunk 1]
```diff
@@ -412,18 +402,15 @@ def test_metadata_server_restart(self):
                     len(errors), 0,
                     "Error stopping agents:\n  {}".format("\n  ".join(errors)))
 
-                # Stop the servers
-                errors = self.stop_servers()
+                # Restart the servers w/o formatting the storage
```
Comment (rpadma2) [inline src/tests/ftest/server/metadata.py:405]: Any reason why we are changing the test steps here? stop_servers and start_servers are going to be different from restart_servers behaviour... Is it not?  Unless restart_servers does the same thing as stop and start servers.

File: src/tests/ftest/util/apricot/apricot/test.py [hunk 2]
```diff
@@ -77,7 +65,11 @@ def __init__(self, *args, **kwargs):
         # Define a test ID using the test_* method name
         self.test_id = self.get_test_name()
 
-        self.test_dir = os.getenv("DAOS_TEST_LOG_DIR", "/tmp")
+        # Define a test unique temporary directory
+        self.base_test_dir = os.getenv("DAOS_TEST_LOG_DIR", "/tmp")
+        self.test_dir = os.path.join(self.base_test_dir, self.test_id)
+        if not os.path.exists(self.test_dir):
```
Comment (rpadma2) [inline src/tests/ftest/util/apricot/apricot/test.py:71]: Under what condition the test directory exists? Restart server?

**Comment by johannlombardi** [comment]
2.0 patch cherry-picked from master in this PR https://github.com/daos-stack/daos/pull/7612

**Pull request closed without merging**